enclave/core/sgx/enter.S

```diff
@@ -105,15 +105,36 @@ oe_enter:
 
     // Abort if SSA[0].GPRSGX.URSP is within the enclave memory range
     cmp %r12, %r9
-    jb .exception_check_pass
+    jb .exception_handler_stack_check
     cmp %r13, %r9
-    jae .exception_check_pass
+    jae .exception_handler_stack_check
     jmp .abort
 
     // Reaching this point implies SSA[0].GPRSGX.RSP is within the enclave
     // memory range so we do not need additional checks.
 
-.exception_check_pass:
+.exception_handler_stack_check:
+    // Stop speculative execution at target of conditional jump
+    lfence
+
+    // Get the exception_handler_stack_check range
+    mov td_exception_handler_stack(%r11), %r14
+    mov td_exception_handler_stack_size(%r11), %r15
```

Comment:
If exception is triggered after exception stack is set, but before stack size is set, then r14 will be valid, but r15 will be zero. This will cause the exception handler to execute with RSP = bottom of exception stack, leading to memory corruption.

Reply:
This seems not an issue as r14 and r15 will not be used to set rsp before the checks here are done.

Reply:
Which are the checks you are referring to?
The checks in line 129 and 131 would both pass.

Reply:
changed the order so the size is set first